Lesson_19/homework_19_1.py

```diff
@@ -0,0 +1,145 @@
+"""Homework # 19.1. Mars Rover “Curiosity” photos."""
+
+# There is an open NASA API that allows you to get JSON data about
+# photos taken by the Curiosity rover on Mars based on certain parameters.
+# Among this data are links to photos that need to be parsed and then,
+# using additional queries, download and save
+# these photos as local files mars_photo1.jpg, mars_photo2.jpg.
+# The task needs to be done using the requests module
+import json
+import logging
+
+import requests
+
+
+def configure_logging():
+    """Logger configuration.
+
+    Returns:
+        logging.Logger: Logging Logger.
+    """
+    logger = logging.getLogger(__name__)
+    logger.setLevel(logging.DEBUG)
+
+    console_handler = logging.StreamHandler()
+    console_handler.setLevel(logging.DEBUG)
+
+    log_formatter = logging.Formatter(
+        ('%(asctime)s.%(msecs)03d : %(module)-12s: '
+         '%(lineno)3d %(levelname)-5s - %(message)s'),
+        '%Y-%m-%d %H:%M:%S',
+    )
+
+    console_handler.setFormatter(log_formatter)
+    logger.addHandler(console_handler)
+
+    return logger
+
+
+_log = configure_logging()
+
+
+class MarsRoverPhotos:
+    """A class to interact with NASA's Mars Rover Photos API.
+
+    Attributes:
+        url (str): The API endpoint URL.
+        params (dict): Parameters to be sent with the API request.
+        timeout (int): Timeout duration for HTTP requests in seconds.
+    """
+
+    def __init__(self, url, params):
+        """Initialize the MarsRoverPhotos instance.
+
+        Args:
+            url (str): The API endpoint URL.
+            params (dict): Parameters to be sent with the API request.
+        """
+        self.url = url
+        self.params = params
+        self.timeout = 5
```

Comment:
this better to set as named init attribute

Reply:
Fixed in commit 31dfb0b758979996cd1a5d4f706a0d589d4b24f9